panel.js

```diff
@@ -5,14 +5,47 @@ var url = document.querySelector('#url');
 var expression = document.querySelector('#expression');
 var followTail = document.querySelector('#checkbox-follow-tail')
 
+function urlMatches(requestUrl, filter) {
+  if (!filter) return true;
+  // RegExp: starts and ends with /
+  if (
+    filter.length > 2 &&
+    filter[0] === "/" &&
+    filter[filter.length - 1] === "/"
+  ) {
+    try {
+      var re = new RegExp(filter.slice(1, -1));
+      return re.test(requestUrl);
+    } catch (e) {
+      return false;
+    }
+  }
+  // Glob: contains * or ?
+  if (filter.includes("*") || filter.includes("?")) {
+    // Escape regex special chars except * and ?
+    var globToRegex = filter
+      .replace(/[.+^${}()|[\]\\]/g, "\\$&")
+      .replace(/\*/g, ".*")
+      .replace(/\?/g, ".");
+    try {
+      var re = new RegExp("^" + globToRegex + "$");
+      return re.test(requestUrl);
+    } catch (e) {
+      return false;
+    }
+  }
+  // Substring
+  return requestUrl.includes(filter);
+}
+
 chrome.devtools.network.onRequestFinished.addListener(request => {
   if (!isJsonType(request)) {
     return;
   }
 
   var urlValue = url.value;
   if (urlValue) {
-    if (!request.request.url.includes(urlValue)) {
+    if (!urlMatches(request.request.url, urlValue)) {
```

Comment:
Since the purpose of the input value has changed, it would be good to change the variable name as well. How about `filterValue`?


```suggestion
var filterValue = url.value;
if (filterValue) {
    if (!urlMatches(request.request.url, filterValue)) {
```